### ISSUE TITLE
Incomplete documents encountered when performing collection query

### ISSUE THREAD

**Issue by sandreae** [p2panda/aquadoggo#348]
When querying a collection of documents (there are 4 in the database) then the field resolver panics as it tries to unwrap a field on one of the documents which _should_ exist (we only accept valid field names in the graphql api) but it doesn't exist and so it panics. 

There is a PR with tests and logging to demonstrate here: #347 

The logs look like this:

```
-------------- TEST START --------------
=== ALL QUERY ROWS RETURNED FROM THE DATABASE ===
[
    QueryRow {
        document_id: "0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253",
        document_view_id: "0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253",
        operation_id: "0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253",
        is_deleted: false,
        cmp_value_cursor: "00de28abe79977d9ccd4ab465a8e9b4000f630d72196e186d6f978f6078c7ca8",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "one",
        value: Some(
            "true",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "00202862d913dae250b40b906f04eb7fb762b4ec5091259536bdff16e56777a57e04",
        document_view_id: "00202862d913dae250b40b906f04eb7fb762b4ec5091259536bdff16e56777a57e04",
        operation_id: "00202862d913dae250b40b906f04eb7fb762b4ec5091259536bdff16e56777a57e04",
        is_deleted: false,
        cmp_value_cursor: "2bd24b8e13f249d2ce4cff0b890be14aa8de0db2a42320b956b8b8cfeac1913e",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "one",
        value: Some(
            "true",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b",
        document_view_id: "0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b",
        operation_id: "0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b",
        is_deleted: false,
        cmp_value_cursor: "3b7c61ff81b1215b89768d0df2b3a5c9b25bd67fb7d56f888b9a3c4f8bbbca8a",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "one",
        value: Some(
            "true",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582",
        document_view_id: "0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582",
        operation_id: "0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582",
        is_deleted: false,
        cmp_value_cursor: "55c6e3be79779d79cfbf621cc0f1da021d5ee383ab1bf259fde4c5c3c7029292",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "one",
        value: Some(
            "true",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b",
        document_view_id: "0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b",
        operation_id: "0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b",
        is_deleted: false,
        cmp_value_cursor: "6edfcb69d1e01125252e97666a5155309d6300563e7cac67134be2862d7a2099",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "two",
        value: Some(
            "false",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "00202862d913dae250b40b906f04eb7fb762b4ec5091259536bdff16e56777a57e04",
        document_view_id: "00202862d913dae250b40b906f04eb7fb762b4ec5091259536bdff16e56777a57e04",
        operation_id: "00202862d913dae250b40b906f04eb7fb762b4ec5091259536bdff16e56777a57e04",
        is_deleted: false,
        cmp_value_cursor: "7e0fc42a246d3ed534de29519318070b65dc6029311186709ac359a414ded47f",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "two",
        value: Some(
            "false",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253",
        document_view_id: "0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253",
        operation_id: "0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253",
        is_deleted: false,
        cmp_value_cursor: "9fc3c2a706a26d268f9c21e1d2ede260d9f1cd4e8ab113c593c3e254878eb382",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "two",
        value: Some(
            "false",
        ),
        field_type: "bool",
        list_index: 0,
    },
    QueryRow {
        document_id: "0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582",
        document_view_id: "0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582",
        operation_id: "0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582",
        is_deleted: false,
        cmp_value_cursor: "a7bd0e16060797b1ffda17e53e2f1717937caa81c127ad0d4d7ca661dba46d3c",
        root_cursor: "",
        is_edited: false,
        owner: OptionalOwner(
            "0000000000000000000000000000000000000000000000000000000000000000",
        ),
        name: "two",
        value: Some(
            "false",
        ),
        field_type: "bool",
        list_index: 0,
    },
]
=== REDUCE ROWS INTO DOCUMENTS ===
Document at row index 2: 0020719a68549439b780e0ab89b64986282b7cb3d3acabbc2fe4c2dc4a77e5ba9253
Fields: ["one"]
Document at row index 4: 0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b
Fields: ["one"]
Document at row index 6: 0020e9b33500f8e8a0530058bc9364c3d4b90c6e1e74e9dadc2a6118ce0d8a93805b
Fields: ["two"]
Final document: 0020b48989f7217c8ca373e8eee32b779b6e8320e1a12525e0b2d16fea76ce834582
Fields: ["two"]
thread 'graphql::queries::collection::tests::prop_test_failure' panicked at 'called `Option::unwrap()` on a `None` value', aquadoggo/src/graphql/resolvers.rs:134:30
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'graphql::queries::collection::tests::prop_test_failure' panicked at 'called `Result::unwrap()` on an `Err` value: reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Ipv4(127.0.0.1)), port: Some(33699), path: "/graphql", query: None, fragment: None }, source: hyper::Error(IncompleteMessage) }', aquadoggo/src/test_utils/client.rs:88:49
```

As you can see, the correct query number of rows (maybe the wrong order?) are returned from the database but when they are reduced into actual documents fields are missing.

**Comment by sandreae** [comment]
May have fixed it, seems the default ordering was set to `None` rather than `Some(Field::Meta(MetaField::DocumentId))`

**Comment by adzialocha** [comment]
The order seems wrong, shouldn't it be: one, two, one, two?

**Comment by adzialocha** [comment]
Maybe it's related to sorting the results by document view id at the end.

**Comment by sandreae** [comment]
> The order seems wrong, shouldn't it be: one, two, one, two?

You mean in the above logs? Yeh, the ordering is wrong there. It works again if the default ordering (as I changed it to now in the PR) is by `DocumentId`. Need a few more tests to make sure this is fixed in other cases, where different orderings are requested. Maybe it actually requires `DocumentViewId`.

**Comment by sandreae** [comment]
Example failure cases over here: https://github.com/p2panda/aquadoggo/pull/354

**Comment by sandreae** [comment]
Ah, I think I see the problem. We're chunking pages by comparing cursor values rather than comparing the document id. I think this is only happening when no other ordering is specified, must have been missed out at some point. I tried to fix it, but then gave up as I feel it might be something you can fix in 3 seconds @adzialocha :sweat_smile:

**Comment by sandreae** [comment]
I think....

**Comment by sandreae** [comment]
Seems fixed by this: https://github.com/p2panda/aquadoggo/pull/354/commits/c91eb21da49ff627054e069079bccd96dab2994c

**Comment by adzialocha** [comment]
> Seems fixed by this: [c91eb21](https://github.com/p2panda/aquadoggo/commit/c91eb21da49ff627054e069079bccd96dab2994c)

Looking good! Can we do a separate PR for each fixed issue? I think its easier to review and understand the effects of these changes.
